actors/builtin/reward/reward_actor_state.go

```diff
@@ -0,0 +1,130 @@
+package reward
+
+import (
+	addr "github.com/filecoin-project/go-address"
+	cid "github.com/ipfs/go-cid"
+	errors "github.com/pkg/errors"
+
+	abi "github.com/filecoin-project/specs-actors/actors/abi"
+	big "github.com/filecoin-project/specs-actors/actors/abi/big"
+	adt "github.com/filecoin-project/specs-actors/actors/util/adt"
+)
+
+type RewardActorState struct {
+	RewardMap cid.Cid // HAMT[Address]AMT[Reward]
+}
+
+type Reward struct {
+	VestingFunction
+	StartEpoch      abi.ChainEpoch
+	EndEpoch        abi.ChainEpoch
+	Value           abi.TokenAmount
+	AmountWithdrawn abi.TokenAmount
+}
+
+type VestingFunction int64
+const (
+	None VestingFunction = iota
+	Linear
```

Comment:
making a note that we might have one more option, more details to come next week cc @jzimmerman